app/controllers/api/v3/imo_callbacks_controller.rb

```diff
@@ -21,6 +24,22 @@ def subscribe
     head :ok
   end
 
+  def read_receipt
+    communication = Communication.find_by!(notification_id: params[:notification_id], communication_type: "imo", detailable_type: "ImoDeliveryDetail")
```

Comment:
I designed this to be based on notification id because the communication/detail aren't created until after the message sends.